index.js

```diff
@@ -201,7 +201,8 @@ RPC.prototype._resolveAndQuery = function (peer, query, cb) {
   dns.lookup(peer.host, function (err, ip) {
     if (err) return cb(err)
     if (self.destroyed) return cb(new Error('k-rpc-socket is destroyed'))
-    self.query({host: ip, port: peer.port}, query, cb)
+    var newPeer = Object.assign({}, peer, {host: ip})
+    self.query(newPeer, query, cb)
```

Comment:
Instead of replacing original `peer` object entirely it creates new object with only `.host` property being changed and leaving other properties as they were passed into `query()` method.

I expect `peer` to contain additional information necessary for establishing WebRTC connection (like SDP) in my use case.